mypy/messages.py

```diff
@@ -997,6 +997,7 @@ def overloaded_signatures_ret_specific(self, index: int, context: Context) -> No
     def operator_method_signatures_overlap(
             self, reverse_class: TypeInfo, reverse_method: str, forward_class: Type,
             forward_method: str, context: Context) -> None:
+        assert reverse_class is not None
```

Comment:
I think we can remove this assert -- we already do a check and an assert up above. It'll also be more consistent this way: we don't really perform checks like these in other similar message methods.

(In fact, it actually ought to be impossible for `reverse_class` to be None at this point since we type-check mypy itself under strict-optional mode: the fact that it sometimes *could* be is an unfortunate consequence of [this cast](https://github.com/python/mypy/blob/master/mypy/nodes.py#L401). I'd rather we eventually fix that TODO so we don't have to keep adding asserts -- but that's definitely very much out-of-scope for this PR.)